front/src/modules/trainschedule/components/ImportTrainSchedule/ImportTrainScheduleTrainsListV2.tsx

```diff
@@ -27,28 +33,79 @@ function LoadingIfSearching({ isLoading, t }: { isLoading: boolean; t: TFunction
 type ImportTrainScheduleTrainsListProps = {
   trainsList: TrainSchedule[];
   rollingStocks: LightRollingStockWithLiveries[];
-  infraId: number;
   isLoading: boolean;
   timetableId: number;
 };
 
 const ImportTrainScheduleTrainsListV2 = ({
   trainsList,
   rollingStocks,
-  infraId,
   isLoading,
   timetableId,
 }: ImportTrainScheduleTrainsListProps) => {
   const { t } = useTranslation(['operationalStudies/importTrainSchedule']);
-  const { openModal } = useModal();
 
   const rollingStockDict = useMemo(
     () => keyBy(rollingStocks, (rollingStock) => rollingStock.name),
     [rollingStocks]
   );
 
+  const { refetch: refetchTimetable } =
+    osrdEditoastApi.endpoints.getTimetableById.useQuerySubscription({ id: timetableId });
+  // const [postPathFinding] = osrdEditoastApi.endpoints.postPathfinding.useMutation();
+  const [postTrainSchedule] = osrdEditoastApi.endpoints.postV2TrainSchedule.useMutation();
+
+  const [status, setStatus] = useState(initialStatus);
+  const dispatch = useAppDispatch();
+
+  async function launchTrainSchedules(payload: TrainScheduleForm) {
+    try {
+      await postTrainSchedule({ body: [payload] }).unwrap();
+    } catch (error) {
+      return false;
+    }
+
+    return true;
+  }
+  async function generateV2TrainSchedules() {
+    const payloads = generateV2TrainSchedulesPayload(trainsList, timetableId);
+
+    const trainsCount = payloads.length;
+    let successfulTrainsCount = 0;
+    // eslint-disable-next-line no-restricted-syntax
+    for await (const payload of payloads) {
+      const success = await launchTrainSchedules(payload);
+      if (success) {
+        successfulTrainsCount += 1;
+      }
+    }
```

Comment:
The function to post a train is very simple, so I think we can remove it without losing in readability:
```suggestion
    let errorsNb = 0;
    // eslint-disable-next-line no-restricted-syntax
   for await (const payload of payloads) {
      try {
        await postTrainSchedule({ body: [payload] }).unwrap();
      } catch (error) {
        errorsNb += 1;
      }
    }
```
You can delete the launchTrainSchedules function and adapt the code to toast the correct success message